mysql/datadog_checks/mysql/statements.py

```diff
@@ -82,17 +84,41 @@ def keyfunc(row):
 
         for row in rows:
             tags = []
-            tags.append('digest:' + row['digest'])
-            if row['schema'] is not None:
-                tags.append('schema:' + row['schema'])
-
-            try:
-                obfuscated_statement = datadog_agent.obfuscate_sql(row['query'])
-            except Exception as e:
-                self.log.warning("Failed to obfuscate query '%s': %s", row['query'], e)
-                continue
-            tags.append('query_signature:' + compute_sql_signature(obfuscated_statement))
-            tags.append('query:' + normalize_query_tag(obfuscated_statement).strip())
+
+            # A NULL digest indicates that some queries are not captured in metrics because
+            # the performance_schema query limit was hit. It is important to treat this row
+            # as a real query so that % time metrics calculations are still valid for all
+            # queries, not just tracked queries.
+            # See: https://dev.mysql.com/doc/refman/5.7/en/statement-summary-tables.html
+            if row['digest'] is None:
+                tags.extend(
+                    [
+                        'digest:' + TAG_VALUE_UNAVAILABLE,
+                        'schema:' + TAG_VALUE_UNAVAILABLE,
+                        'query:' + TAG_VALUE_UNAVAILABLE,
+                        'query_signature:' + TAG_VALUE_UNAVAILABLE,
+                    ]
+                )
+                self.log.warning(
+                    'Encountered a null digest row in performance_schema.events_statements_summary_by_digest. '
+                    'This indicates that not all queries could be tracked. You may wish to raise the system '
+                    'variable `performance-schema-digests-size` or truncate this table periodically.'
+                )
+            else:
+                tags.append('digest:' + row['digest'])
+
+                if row['schema'] is not None:
+                    tags.append('schema:' + row['schema'])
+
+                try:
+                    obfuscated_statement = datadog_agent.obfuscate_sql(row['query'])
+                except Exception as e:
+                    obfuscated_statement = TAG_VALUE_UNAVAILABLE
```

Comment:
Should we be re-using `unavailable` for obfuscation errors? We'd be confusing obfuscation errors with events_statements truncation problems.

Reply:
Yeah makes sense, I'll make it `error`